### PR TITLE
Add IP2WHOIS

### PR DESCRIPTION
**Pull request by ip2location**
# Description

Add IP2WHOIS observer analyzer.

## Type of change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality).

# Checklist

- [x] I have read and understood the rules about [how to Contribute](https://intelowl.readthedocs.io/en/latest/Contribute.html) to this project
- [x] The pull request is for the branch `develop`
- [x] A new plugin (analyzer, connector, visualizer, playbook, pivot or ingestor) was added or changed, in which case:
    - [x] I strictly followed the documentation ["How to create a Plugin"](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-add-a-new-plugin)
    - [x] [Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Usage.md) file was updated.
    - [ ] [Advanced-Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Advanced-Usage.md) was updated (in case the plugin provides additional optional configuration).
    - [x] If the plugin requires mocked testing, `_monkeypatch()` was used in its class to apply the necessary decorators.
    - [x] I have dumped the configuration from Django Admin using the `dumpplugin` command and added it in the project as a data migration. (["How to share a plugin with the community"](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-share-your-plugin-with-the-community))
    - [ ] If a File analyzer was added and it supports a mimetype which is not already supported, you added a sample of that type inside the archive `test_files.zip` and you added the default tests for that mimetype in [test_classes.py](https://github.com/intelowlproject/IntelOwl/blob/master/tests/analyzers_manager/test_classes.py).
    - [ ] If you created a new analyzer and it is free (does not require API keys), please add it in the `FREE_TO_USE_ANALYZERS` playbook by following [this guide](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-modify-a-plugin).
    - [x] Check if it could make sense to add that analyzer/connector to other [freely available playbooks](https://intelowl.readthedocs.io/en/develop/Usage.html#list-of-pre-built-playbooks).
    - [x] I have provided the resulting raw JSON of a finished analysis and a screenshot of the results. 
    - [ ] I have added that raw JSON sample to the `MockUpResponse` of the `_monkeypatch()` method. This serves us to provide a valid sample for testing.
- [ ] If external libraries/packages with restrictive licenses were used, they were added in the [Legal Notice](https://github.com/certego/IntelOwl/blob/master/.github/legal_notice.md) section.
- [x] Linters (`Black`, `Flake`, `Isort`) gave 0 errors. If you have correctly installed [pre-commit](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-start-setup-project-and-development-instance), it does these checks and adjustments on your behalf.
- [ ] I have added tests for the feature/bug I solved (see `tests` folder). All the tests (new and old ones) gave 0 errors.
- [ ] If changes were made to an existing model/serializer/view, the docs were updated and regenerated (check [CONTRIBUTE.md](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Contribute.md)).
- [ ] If the GUI has been modified:
    - [ ] I have a provided a screenshot of the result in the PR.
    - [ ] I have created new frontend tests for the new component or updated existing ones.
- [X] After you had submitted the PR, if `DeepSource`, `Django Doctors` or other third-party linters have triggered any alerts during the CI checks, I have solved those alerts.

### Important Rules
- If you miss to compile the Checklist properly, your PR won't be reviewed by the maintainers.
- Everytime you make changes to the PR and you think the work is done, you should explicitly ask for a review. After being reviewed and received a "change request", you should explicitly ask for a review again once you have made the requested changes.
intelowl-ip2whois-exampleoutput
[ip2whois_exampleoutput_msncom.json](https://github.com/intelowlproject/IntelOwl/files/15185196/ip2whois_exampleoutput_msncom.json)

### CODE REVIEW

File: api_app/analyzers_manager/observable_analyzers/ip2whois.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        response = location_info.json()
+        return response
+
+    @classmethod
+    def _monkeypatch(cls):
+        patches = [
+            if_mock_connections(
+                patch(
+                    "requests.get",
+                    return_value=MockUpResponse({}, 200),
```
Comment (mlodic) [inline api_app/analyzers_manager/observable_analyzers/ip2whois.py:40]: you could put in this json here a sample of a possible output to make the test more real and to have a record of the expected results keys

**Review by mlodic — COMMENTED** [review]
cool, thank you for the contribution. Just a little change requested. Then please check the CI results like tests and deepsource and do the required adjusts. :)

File: api_app/analyzers_manager/observable_analyzers/ip2whois.py (hunk 1 above)
Comment (ip2location) [reply api_app/analyzers_manager/observable_analyzers/ip2whois.py:40]: Hi @mlodic , thanks for the comment, we will try to add one example now. Meanwhile, can we get advise from you on how to solve the errors reported by the backend-tests and DeepSource: Python?

Comment (mlodic) [reply api_app/analyzers_manager/observable_analyzers/ip2whois.py]: yep it works, just now the problem is with the linters, you should apply black and flake as stated in the docs

Comment (ip2location) [reply api_app/analyzers_manager/observable_analyzers/ip2whois.py]: Hi @mlodic , we have pushed some fixes for the issues, let's wait and see.

Comment (ip2location) [reply api_app/analyzers_manager/observable_analyzers/ip2whois.py]: Hi @mlodic , I found that the Build & Tests / backend-tests is failing, can you please help us to see if the issue related to our codes?

**Pull request merged**